### DISCUSSION TITLE
Stop & go for specific URL on Proxy mode.

### DISCUSSION THREAD

**Discussion by Jang-Jian** [bluenviron/mediamtx#1926] [Questions]
Hi, developer,

The question is described as this:
I have a two URL and set in single yml file:

paths:
  proxied1:
    source: rtsp://url1

  proxied2:
    source: rtsp://url1

Do you have any ideas to switch these scenario by command or code(api)? Not disable in yml file.

scenario 1. 'stop proxied1'  and 'execute proxied2' .
scenario 2. 'execute proxied1'  and 'stop proxied2' .
scenario 3. 'execute proxied1'  and 'execute proxied2' .

Thank you to your reply !

**Comment by aler9** [comment]
Hello, you can either:

1. prepare 3 different configuration file, each with a scenario, call them mediamtx.scenario1.yml, mediamtx.scenario2.yml, mediamtx.scenario3.yml. Then write a a script that once called copies a certain configuration into mediamtx.yml. The server is able to detect changes to mediamtx.yml and will apply the new configuration.

2. use the API to dynamically add or remove paths proxied1 and proxied2 by using [/v2/config/paths/add/{name}](https://bluenviron.github.io/mediamtx/#operation/configPathsAdd) and and [/v2/config/paths/remove/{name}](https://bluenviron.github.io/mediamtx/#operation/configPathsRemove)

**Reply by Jang-Jian** [reply]
Hello, developer,

I appreciate your reply!

For your suggestion 2., I have some questions:

1. Do you have any example(code) about this part? Maybe your repository has these examples, but I haven't find them yet !

2. For suggestion 1., you said 'The server is able to detect changes to mediamtx.yml and will apply the new configuration.' So behavior of suggestion 2. is also like suggestion 1. ?


Thank you to your reply again!

**Reply by aler9** [reply]
The two suggestions result in an identical outcome and are both easy to code. If you need an example, look for examples on how to copy a file or call an HTTP endpoint for your chosen programming language, since they are two of the most common operations of every programming language. I can't add examples since everyone has its own language of choice. Regarding the API, the full documentation is here: https://bluenviron.github.io/mediamtx/

**Reply by Jang-Jian** [reply]
Got it ! 
This is a very useful information.
Thank you !

[a comment by an automated account is not shown]
